src/Umbraco.Infrastructure/Migrations/Install/DatabaseDataCreator.cs

```diff
@@ -1745,14 +1747,104 @@ private void CreatePropertyTypeData()
+            }
+        }
+    }
+
+    private bool TryCreateCulture(string isoCode, [NotNullWhen(true)] out CultureInfo? culture)
+    {
+        try
+        {
+            culture = CultureInfo.GetCultureInfo(isoCode);
+            return true;
+        }
+        catch (CultureNotFoundException ex)
+        {
+            _logger.LogWarning(ex, "CultureInfo could not be created because culture '{IsoCode}' is not available. The language will not be created.", isoCode);
+            culture = null;
+            return false;
+        }
+    }
+
+    private void CreateLanguageDataOld()
+    {
+        var defaultCulture = CultureInfo.GetCultureInfo("en-US");
+        bool createdDefaultLanguage = ConditionalInsert(
             Constants.Configuration.NamedOptions.InstallDefaultData.Languages,
             "en-us",
-            new LanguageDto { Id = 1, IsoCode = "en-US", CultureName = "English (United States)", IsDefault = true },
+            new LanguageDto { Id = 1, IsoCode = defaultCulture.Name, CultureName = defaultCulture.EnglishName, IsDefault = true },
             Constants.DatabaseSchema.Tables.Language,
             "id");
 
+        // For languages we support the installation of records that are additional to the default installed data.
+        // We can do this as they are specified by ISO code, which is enough to fully detail them. All other customizable install data is
+        // specified by GUID, and hence we only know about the set that are installed by default).
+        InstallDefaultDataSettings? languageInstallDefaultDataSettings = _installDefaultDataSettings.Get(Constants.Configuration.NamedOptions.InstallDefaultData.Languages);
+        if (languageInstallDefaultDataSettings?.InstallData == InstallDefaultDataOption.Values)
+        {
+            CreateSpecifiedlLanguageData(languageInstallDefaultDataSettings.Values, createdDefaultLanguage, defaultCulture.Name);
+        }
+    }
+
+    private void CreateSpecifiedlLanguageData(IList<string> isoCodes, bool createdDefaultLanguage, string defaultIsoCode)
+    {
+        // Create language records for all specified languages.
+        // We omit en-US if specified, as that will have been created as the "default default" language already.
+        // We set the first provided language as the default if we haven't already created one.
+        var isFirstConfiguredLanguage = true;
+        foreach (var isoCode in isoCodes.Where(x => !string.Equals(x, defaultIsoCode, StringComparison.OrdinalIgnoreCase)))
+        {
+            var culture = new CultureInfo(isoCode);
+            var dto = new LanguageDto
+            {
+                IsoCode = culture.Name,
+                CultureName = culture.DisplayName,
+                IsDefault = !createdDefaultLanguage && isFirstConfiguredLanguage
+            };
+            _database.Insert(Constants.DatabaseSchema.Tables.Language, "id", true, dto);
+            isFirstConfiguredLanguage = false;
+        }
+    }
+
```

Comment:
Just a friendly reminder that this still needs to be cleaned up. The return type on `ConditionalInsert` isn't used anymore either and since it's private, we can easily add it back later if we have a need for it...

Reply:
Oops... meant to remove that before committing of course!  Have tidied up now.